hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/security/acl/OzoneAuthorizerFactory.java

```diff
@@ -78,6 +80,13 @@ private static IAccessAuthorizer create(
       return configure(authorizer, om, km, pm);
     }
 
+    // If authorizer isn't native and shareable tmp dir is enabled,
+    // then return the shared tmp hybrid authorizer.
+    if (conf.getBoolean(OZONE_OM_ENABLE_OFS_SHARED_TMP_DIR,
+        OZONE_OM_ENABLE_OFS_SHARED_TMP_DIR_DEFAULT)) {
+      return new SharedTmpDirAuthorizer(om, conf);
+    }
+
     final IAccessAuthorizer authorizer = newInstance(clazz, conf);
     return authorizer instanceof OzoneNativeAuthorizer
         ? configure((OzoneNativeAuthorizer) authorizer, om, km, pm)
```

Comment:
By reording the conditions and passing ready-to-use instances to the hybrid one, we can get rid of the code duplicated in `SharedTmpDirAuthorizer` from this class.

In addition to the suggested change, `configure()` needs to be updated to return `OzoneNativeAuthorizer`.

```suggestion
    final IAccessAuthorizer authorizer = newInstance(clazz, conf);
    
    if (authorizer instanceof OzoneNativeAuthorizer) {
      return configure((OzoneNativeAuthorizer) authorizer, om, km, pm);
    }
    
    // If authorizer isn't native and shareable tmp dir is enabled,
    // then return the shared tmp hybrid authorizer.
    if (conf.getBoolean(OZONE_OM_ENABLE_OFS_SHARED_TMP_DIR,
        OZONE_OM_ENABLE_OFS_SHARED_TMP_DIR_DEFAULT)) {
      return new SharedTmpDirAuthorizer(
        configure(new OzoneNativeAuthorizer(), om, km, pm),
        authorizer);
    }
    
    return authorizer;
```